contracts/base/ERC721Drop.sol

```diff
@@ -0,0 +1,272 @@
+    function collectPriceOnClaim(
+        uint256 _quantityToClaim,
+        address _currency,
+        uint256 _pricePerToken
+    ) internal override(DropSinglePhase, ERC721SignatureMint) {
+        if (_pricePerToken == 0) {
+            return;
+        }
+
+        uint256 totalPrice = _quantityToClaim * _pricePerToken;
+
+        if (_currency == CurrencyTransferLib.NATIVE_TOKEN) {
+            if (msg.value != totalPrice) {
+                revert ERC721Drop__MustSendTotalPrice(msg.value, totalPrice);
+            }
+        }
+
+        CurrencyTransferLib.transferCurrency(_currency, msg.sender, primarySaleRecipient(), totalPrice);
+    }
+
+    /// @dev Transfers the NFTs being claimed.
+    function transferTokensOnClaim(address _to, uint256 _quantityBeingClaimed)
+        internal
+        override
+        returns (uint256 startTokenId)
+    {
+        startTokenId = nextTokenIdToMint();
+        _safeMint(_to, _quantityBeingClaimed);
+    }
+
+    /// @dev Checks whether primary sale recipient can be set in the given execution context.
+    function _canSetPrimarySaleRecipient() internal view override returns (bool) {
+        return msg.sender == owner();
+    }
+
+    /// @dev Checks whether owner can be set in the given execution context.
+    function _canSetOwner() internal view override returns (bool) {
+        return msg.sender == owner();
+    }
+
+    /// @dev Checks whether royalty info can be set in the given execution context.
+    function _canSetRoyaltyInfo() internal view override returns (bool) {
+        return msg.sender == owner();
+    }
+
+    /// @dev Checks whether contract metadata can be set in the given execution context.
+    function _canSetContractURI() internal view override returns (bool) {
+        return msg.sender == owner();
+    }
+
+    /// @dev Checks whether platform fee info can be set in the given execution context.
+    function _canSetClaimConditions() internal view override returns (bool) {
+        return msg.sender == owner();
+    }
+
+    /// @dev Returns whether lazy minting can be done in the given execution context.
+    function _canLazyMint() internal view virtual override returns (bool) {
+        return msg.sender == owner();
+    }
+
+    /// @dev Checks whether NFTs can be revealed in the given execution context.
+    function _canReveal() internal view virtual returns (bool) {
+        return msg.sender == owner();
+    }
+
+    /*///////////////////////////////////////////////////////////////
+                        Miscellaneous
+    //////////////////////////////////////////////////////////////*/
+
+    function _dropMsgSender() internal view virtual override returns (address) {
+        return msg.sender;
+    }
+
+    function mint(
+        address,
+        uint256,
+        string memory,
+        bytes memory
+    ) public virtual override {
+        revert("Not authorized to mint.");
```

Comment:
might be better to revert with "not implemented, use lazyMint instead" or something like that, otherwise ppl might get confused